spec/lucky_web/text_helpers_spec.cr

```diff
@@ -0,0 +1,294 @@
+  end
+
+  it "highlights pending" do
+    view.highlight("   ", "blank text is returned verbatim").should eq "   "
+  end
+
+  it "highlights returning blank string for nil" do
+    view.highlight(nil, "blank string is returned for nil").should eq ""
+  end
+
+  it "highlights with regexp" do
+    view.highlight("This is a beautiful! morning", "beautiful!").should eq "This is a <mark>beautiful!</mark> morning"
+    view.highlight("This is a beautiful! morning", "beautiful! morning").should eq "This is a <mark>beautiful! morning</mark>"
+    view.highlight("This is a beautiful? morning", "beautiful? morning").should eq "This is a <mark>beautiful? morning</mark>"
+  end
+
+  it "highlights accepts regexp" do
+    view.highlight("This day was challenging for judge Allen and his colleagues.", /\ballen\b/i).should eq "This day was challenging for judge <mark>Allen</mark> and his colleagues."
+  end
+
+  it "highlights with multiple phrases in one pass" do
+    view.highlight("wow em", %w(wow em), highlighter: "<em>\\1</em>").should eq %(<em>wow</em> <em>em</em>)
+  end
+
+  it "highlights with html" do
+    view.highlight("<p>This is a beautiful morning, but also a beautiful day</p>", "beautiful").should eq "<p>This is a <mark>beautiful</mark> morning, but also a <mark>beautiful</mark> day</p>"
+    view.highlight("<p>This is a <em>beautiful</em> morning, but also a beautiful day</p>", "beautiful").should eq "<p>This is a <em><mark>beautiful</mark></em> morning, but also a <mark>beautiful</mark> day</p>"
+    view.highlight("<p>This is a <em class=\"error\">beautiful</em> morning, but also a beautiful <span class=\"last\">day</span></p>", "beautiful").should eq "<p>This is a <em class=\"error\"><mark>beautiful</mark></em> morning, but also a <mark>beautiful</mark> <span class=\"last\">day</span></p>"
+    view.highlight("<p class=\"beautiful\">This is a beautiful morning, but also a beautiful day</p>", "beautiful").should eq "<p class=\"beautiful\">This is a <mark>beautiful</mark> morning, but also a <mark>beautiful</mark> day</p>"
+    view.highlight("<p>This is a beautiful <a href=\"http://example.com/beautiful\#top?what=beautiful%20morning&when=now+then\">morning</a>, but also a beautiful day</p>", "beautiful").should eq "<p>This is a <mark>beautiful</mark> <a href=\"http://example.com/beautiful\#top?what=beautiful%20morning&when=now+then\">morning</a>, but also a <mark>beautiful</mark> day</p>"
+    view.highlight("<div>abc div</div>", "div", highlighter: "<b>\\1</b>").should eq "<div>abc <b>div</b></div>"
+  end
+
+  it "highlights with block" do
+    view.highlight("one two three", ["one", "two", "three"]) { |word| "<b>#{word}</b>" }.should eq "<b>one</b> <b>two</b> <b>three</b>"
+    view.test_highlight.should eq "This is a <a href=\"#beautiful\">beautiful</a> morning, but also a <a href=\"#beautiful\">beautiful</a> day"
+  end
```

Comment:
Could you group the different type of methods in a `describe` block? Like `describe "highlighting/truncating/cycling/etc."? I think that might aid readability